frontend/alert/components/AlertForm/index.tsx

```diff
@@ -184,10 +211,15 @@ const AlertForm = ({
         Promise.allSettled(promises)
             .then((responses) => responses.forEach(
                 (res: PromiseSettledResult<Response>, i) => {
+                    //fulfilled if response is returned, even if reg is unsuccessful.
                     if (res.status === "fulfilled") {
-                        console.log("fulfilled")
-                        setResponse(res.value);
-                        deselectCourse(sections[i]);
+                        if (res.value.status === 200) {
+                            setResponse(res.value);
+                            deselectCourse(sections[i]);
+                        } else {
+                            setResponse(res.value);
+                        }
+                    //only if error occurred
```

Comment:
could say "only if network error occured" (see stack overflow post above) to disambiguate with non-2xx responses, which might be confused with an "error"